Lib/pathlib.py

```diff
@@ -92,6 +108,26 @@ def _compile_pattern(pat, case_sensitive):
     return re.compile(fnmatch.translate(pat), flags).match
 
 
+@functools.lru_cache()
+def _compile_pattern_lines(pattern_lines, case_sensitive):
+    parts = ['^']
+    for part in pattern_lines.splitlines(keepends=True):
+        if part == '**\n':
+            part = r'[\s\S]*^'
```

Comment:
I assume the `[\s\S]` pattern implies something slightly more nuanced than `.`? Supposedly the two classes are the opposite of each other, and so should include everything.

A brief comment here will save the question being asked every time. (You can't over-explain your regex!)

Reply:
Added a chonky comment! The important bit is that we're **not** running in `re.DOTALL` mode, and so `.` does not match newlines, which in our case represent path separators. It's this trick that allows `*` wildcards to not match path separators.